manim/utils/color/core.py

```diff
@@ -519,6 +521,39 @@ def invert(self, with_alpha=False) -> ManimColor:
         """
         return ManimColor(1.0 - self._internal_value, with_alpha)
 
+    @overload
+    def opacity(self, opacity: float) -> ManimColor:
+        """Returns a new ManimColor with the same color and the given opacity
+
+        Parameters
+        ----------
+        opacity : float
+            The opacity for the new ManimColor
+
+        Returns
+        -------
+        ManimColor
+            The new ManimColor object with changed opacity
+        """
+
+    @overload
+    def opacity(self, opacity: None) -> float:
+        """Returns the opacity of the current ManimColor in a range from zero to one
+
+        Returns
+        -------
+        float
+            The opacity of the ManimColor
+        """
+
+    def opacity(self, opacity=None):
+        """Returns a new ManimColor with the same color and a new opacity or changes the opacity"""
+        if opacity is None:
+            return self._internal_value[3]
+        tmp = self._internal_value.copy()
+        tmp[3] = opacity
+        return ManimColor.parse(tmp)
```

Comment:
I'm not sure about this overload. Could it be, instead, something like `ManimColor.with_opacity(opacity)` to return a new `ManimColor` with that different opacity, and `ManimColor.get_opacity()` or `ManimColor.opacity` as a property to get the current opacity?
```suggestion
    def with_opacity(self, opacity: float) -> ManimColor:
        """Returns a new ManimColor with the same color and the given opacity.

        Parameters
        ----------
        opacity
            The opacity for the new ManimColor.

        Returns
        -------
        ManimColor
            The new ManimColor object with changed opacity.
        """
        tmp = self._internal_value.copy()
        tmp[3] = opacity
        return ManimColor.parse(tmp)

    @property
    def opacity(self) -> float:
        """Returns the opacity of the current ManimColor, in a range from zero to one.

        Returns
        -------
        float
            The opacity of the ManimColor.
        """
        return self._internal_value[3]
```